examples/demo13.py

```diff
@@ -0,0 +1,378 @@
+        # Even when the detector is unexposed to any radiation, the electron-hole pairs that
+        # are generated within the depletion region due to finite temperature are swept by the
+        # high electric field at the junction of the photodiode. This small reverse bias
+        # leakage current is referred to as 'Dark current'. It is specified by the average
+        # number of electrons reaching the detectors per unit time and has an associated
+        # Poisson noise since it is a random event.
+        dark_img = galsim.ImageF(bounds=final_image.bounds,
+                                 init_value=wfirst.dark_current*wfirst.exptime)
+        dark_img.addNoise(poisson_noise)
+        final_image += dark_img
+
+        # NOTE: Sky level and dark current might appear like a constant background that can be
+        # simply subtracted. However, these contribute to the shot noise and matter for the
+        # non-linear effects that follow. Hence, these must be included at this stage of the 
+        # image generation process. We subtract these backgrounds in the end.
+
+        # 3) Applying a quadratic non-linearity:
+        # In order to convert the units from electrons to ADU, we must multiply the image by a
+        # gain factor. The gain has a weak dependency on the charge present in each pixel. This
+        # dependency is accounted for by changing the pixel values (in electrons) and applying
+        # a constant nominal gain later, which is unity in our demo.
+
+        # Save the image before applying the transformation to see the difference:
+        final_image_3 = final_image.copy()
+
+        NLfunc = wfirst.NLfunc        # a quadratic non-linear function
+        final_image.applyNonlinearity(NLfunc=NLfunc)
+        logger.debug('Applied nonlinearity to {0}-band image'.format(filter_name))
+        final_image_4 = final_image.copy()
+        # Isolate the changes due to non-linear gain.
+        diff = final_image_4-final_image_3
+
+        out_filename = os.path.join(outpath,'demo13_NL_{0}.fits'.format(filter_name))
+        final_image_4.write(out_filename)
+        out_filename = os.path.join(outpath,'demo13_diff_NL_{0}.fits'.format(filter_name))
+        diff.write(out_filename)
+
+        # 4) Including Interpixel capacitance:
+        # The voltage read at a given pixel location is influenced by the charges present in
+        # the neighboring pixel locations due to capacitive coupling of sense nodes. This
+        # interpixel capacitance effect is modeled as a linear effect that is described as a
+        # convolution of a 3x3 kernel with the image. The WFIRST kernel is not normalized to
+        # have the entries add to unity and hence must be normalized inside the routine.
+
+        final_image.applyIPC(IPC_kernel=wfirst.ipc_kernel,edge_treatment='extend',
+                             kernel_normalization=True)
+        # Here, we use `edge_treatment='extend'`, which pads the image with zeros before
```

Comment:
This is the default.  Recommend not mentioning `edge_treatment` at all, since the default is what most people will want to do most of the time.  I think explaining it here is more confusing than instructive.